### PR TITLE
nvprims native batch norm patch

### PR DESCRIPTION
**Pull request by jjsjann123**
Cherry-picking: https://github.com/csarofeen/pytorch/pull/2104

- [x] Added explicit cast on inputs to nvprims.native_batch_norm. This avoids the explicit cast, which gives us issue on fusion definition.
- [x] add python repro with dynamo

### CODE REVIEW

**Comment by jjsjann123** [comment]
Failing test looks real... I need to fix this one.

**Review by mruberry — APPROVED** [review]
Stamped! (So long as @kevinstephano and @IvanYashchuk are happy)

File: test/test_nvfuser_dynamo.py [hunk 1]
```diff
@@ -45,6 +45,25 @@ def func(a, b):
...
+        w = make_tensor((3), device="cuda", dtype=torch.float32)
+        b = make_tensor((3), device="cuda", dtype=torch.float32)
+
+        @torchdynamo.optimize("nvprims_nvfuser")
+        def func(mat1, mat2, w, b):
+            o = torch.matmul(mat1, mat2)
+            return torch.batch_norm(o, w, b, None, None, True, 1e-2, 1e-5, True)
+
+        # No warnings and no errors
+        with torch.cuda.amp.autocast():
```
Comment (IvanYashchuk) [inline test/test_nvfuser_dynamo.py:60]: Does it do the expected thing when `autocast` is outside of `@torchdynamo.optimize("nvprims_nvfuser")`? I hope Dynamo doesn't suppress the `autocast` context.

Comment (IvanYashchuk) [reply test/test_nvfuser_dynamo.py:60]: Cool!

**Review by IvanYashchuk — APPROVED** [review]
Merge it!
